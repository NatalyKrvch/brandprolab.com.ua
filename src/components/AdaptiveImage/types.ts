export interface PictureSource {
  srcSet: string;
  media: string;
}

export interface AdaptiveImageProps {
  sources: PictureSource[];
  fallbackSrc: string | null;
  alt: string;
  className?: string;
}
