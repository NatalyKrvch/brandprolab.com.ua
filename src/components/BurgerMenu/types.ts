import type { MenuItemType } from '../NavigationMenu';
import type { SocialIcon } from '../SocialIcons';

export interface BurgerMenuProps {
  isOpen: boolean;
  onClose: () => void;
  photoUrl: string;
  socialLinks: SocialIcon[];
  menuItems: MenuItemType[];
  panelClassName?: string;
  navClassName?: string;
  animationMs?: number;
  portalId?: string;
}
