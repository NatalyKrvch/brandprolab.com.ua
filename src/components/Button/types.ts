export interface ButtonProps {
  onClick: () => void;
  color?: 'white' | 'teal';
}
