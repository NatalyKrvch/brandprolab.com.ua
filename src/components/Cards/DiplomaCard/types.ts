export interface DiplomaCardProps {
  diplomaURL: string;
  onClick?: () => void;
}
