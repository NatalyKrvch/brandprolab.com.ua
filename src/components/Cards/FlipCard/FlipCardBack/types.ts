export interface FlipCardBackProps {
  header: string;
  list: string[];
}
