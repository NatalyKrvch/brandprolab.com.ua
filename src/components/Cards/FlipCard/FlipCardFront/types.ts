export interface FlipCardFrontProps {
  iconURL: string;
  header: string;
  label: string;
  description: string;
  callToAction: string;
}
