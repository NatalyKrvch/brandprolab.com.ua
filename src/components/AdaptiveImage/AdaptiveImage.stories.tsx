import type { Meta, StoryObj } from '@storybook/react';

import AdaptiveImage from './AdaptiveImage';
import { AdaptiveImageProps } from './types';

const meta: Meta<typeof AdaptiveImage> = {
  title: 'Components/AdaptiveImage',
  component: AdaptiveImage,
  parameters: {
    controls: { expanded: true },
  },
  argTypes: {
    alt: {
      control: 'text',
      description: 'Alt text for the image.',
      table: { type: { summary: 'string' } },
    },
    className: {
      control: 'text',
      description: 'CSS classes applied to the picture and img elements.',
      table: { type: { summary: 'string' } },
    },
    fallbackSrc: {
      control: 'text',
      description:
        'Fallback image URL for browsers that do not support <picture>.',
      table: { type: { summary: 'string' } },
    },
    sources: {
      control: 'object',
      description: 'Array of source objects with srcSet and media properties.',
      table: {
        type: { summary: 'Array<{ srcSet: string; media: string; }>' },
      },
    },
  },
};

export default meta;

type Story = StoryObj<AdaptiveImageProps>;

export const Default: Story = {
  args: {
    alt: 'Example image',
    className: 'max-w-full h-auto rounded-lg',
    fallbackSrc: '/images/fallback.jpg',
    sources: [
      { srcSet: '/images/temp/photo_mobile.png', media: '(max-width: 600px)' },
      {
        srcSet: '/images/temp/photo_tablet.png',
        media: '(min-width: 601px) and (max-width: 1200px)',
      },
      {
        srcSet: '/images/temp/photo_desktop.png',
        media: '(min-width: 1201px)',
      },
    ],
  },
};

export const CustomClass: Story = {
  args: {
    ...Default.args,
    className: 'border border-gray-300 p-2 shadow-lg',
  },
};

export const NoSources: Story = {
  args: {
    alt: 'No sources example',
    className: '',
    fallbackSrc: '/images/fallback-only.jpg',
    sources: [],
  },
};
