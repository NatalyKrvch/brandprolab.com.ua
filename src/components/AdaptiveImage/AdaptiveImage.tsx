import clsx from 'clsx';

import type { AdaptiveImageProps } from './types';

export default function AdaptiveImage({
  sources,
  fallbackSrc,
  className,
  alt = '',
}: AdaptiveImageProps) {
  if (!fallbackSrc) return null;

  if (sources.length > 0) {
    return (
      <picture className={clsx(className)}>
        {sources.map(srcObj => (
          <source
            key={srcObj.media}
            srcSet={srcObj.srcSet}
            media={srcObj.media}
          />
        ))}

        <img src={fallbackSrc} alt={alt} className={clsx(className)} />
      </picture>
    );
  }
}
