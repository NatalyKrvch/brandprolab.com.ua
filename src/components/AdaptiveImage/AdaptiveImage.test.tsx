import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import AdaptiveImage from './AdaptiveImage';
import { AdaptiveImageProps } from './types';

describe('AdaptiveImage component', () => {
  const defaultProps: AdaptiveImageProps = {
    alt: 'Test Alt',
    sources: [
      { srcSet: '/img-small.jpg', media: '(max-width: 600px)' },
      { srcSet: '/img-large.jpg', media: '(min-width: 601px)' },
    ],
    fallbackSrc: '/fallback.jpg',
    className: 'test-class',
  };

  it('returns null when no fallbackSrc is provided', () => {
    const { container, asFragment } = render(
      // @ts-expect-error testing missing fallbackSr
      <AdaptiveImage alt="X" sources={defaultProps.sources} />,
    );
    expect(container.firstChild).toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });

  it('returns nothing when sources array is empty', () => {
    const props = { ...defaultProps, sources: [] };
    const { container, asFragment } = render(<AdaptiveImage {...props} />);
    expect(container.firstChild).toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a <picture> with multiple <source> and a fallback <img>', () => {
    const { container, asFragment } = render(
      <AdaptiveImage {...defaultProps} />,
    );

    const picture = container.querySelector('picture');
    expect(picture).toBeInTheDocument();
    expect(picture).toHaveClass('test-class');

    const sources = container.querySelectorAll('source');
    expect(sources).toHaveLength(defaultProps.sources.length);
    expect(sources[0]).toHaveAttribute('srcset', '/img-small.jpg');
    expect(sources[0]).toHaveAttribute('media', '(max-width: 600px)');
    expect(sources[1]).toHaveAttribute('srcset', '/img-large.jpg');
    expect(sources[1]).toHaveAttribute('media', '(min-width: 601px)');

    const img = screen.getByAltText('Test Alt') as HTMLImageElement;
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', '/fallback.jpg');
    expect(img).toHaveClass('test-class');

    expect(asFragment()).toMatchSnapshot();
  });
});
