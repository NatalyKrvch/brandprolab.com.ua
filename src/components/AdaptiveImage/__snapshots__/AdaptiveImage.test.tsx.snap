// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdaptiveImage component renders a <picture> with multiple <source> and a fallback <img> 1`] = `
<DocumentFragment>
  <picture
    class="test-class"
  >
    <source
      media="(max-width: 600px)"
      srcset="/img-small.jpg"
    />
    <source
      media="(min-width: 601px)"
      srcset="/img-large.jpg"
    />
    <img
      alt="Test Alt"
      class="test-class"
      src="/fallback.jpg"
    />
  </picture>
</DocumentFragment>
`;

exports[`AdaptiveImage component returns nothing when sources array is empty 1`] = `<DocumentFragment />`;

exports[`AdaptiveImage component returns null when no fallbackSrc is provided 1`] = `<DocumentFragment />`;
